### DISCUSSION TITLE
Some information about compiling the printerf version AviSynthCUDAFilters

### DISCUSSION THREAD

**Discussion by Subterrestrial** [pinterf/AviSynthCUDAFilters#1] [Ideas]
Amateur as I am, I would like to share some information about building the printerf version AviSynthCUDAFilters because I read a reply referred that there were problems in compling these filters with AVS+3.7.2 (https://github.com/AviSynth/AviSynthPlus/issues/296).
  When I compiled printerf's AviSynthCUDAFilters with AVS+3.7.0 by CUDA11 compiler and run the program using a pascal architecture card(GTX10xx series, just like the one which printerf got when he moded the soure file), it worked very well. But when I changed the card to a turing architecture card(GTX16xx and RTX20xx) and comiled the file again with AVS+3.7.2, the filter failed to work. It seemed the graphic gard memory and cpu memory failed to synchronise. The same problem was also reported in a Japanese forum where there is special column discussing Avisynth topics([url]https://toro.2ch.sc/test/read.cgi/avi/1653113801/l50[/url]).

  Anyway, I finally figured out a way to make it work: using CUDA 8 compiler to compile printerf's AviSynthCUDAFilters, in this way I could make the program run on a turing architecture card but I can't guarantee it can run on ampere architecture card(RTX30xx) because I don't have a card for testing. However, it seems that the program may run on ampere architecture cards since Nekopanda's program was compiled by CUDA8 compiler and the program was reported to work on those cards ([url]https://mevius.5ch.net/test/read.cgi/avi/1666875696/[/url]).

  Sorry but there are some off topic things I want to add. Actually the source code that Nekopanda used to compile his last version cuda filters which is in his repository of Amtsukaze project was not updated to his repository of AviSynthCUDAFilters project. The date in which the program file was comiled is about half year later than the date of the final change in his AviSynthCUDAFilters project. Another decisive evidence is that a funtion in printerf's AviSynthCUDAFilters is lacking a parameter which resulted in KFMDeint.avsi in Amtsukaze failing directly applied in(bool "is120" in function KFMDeint). Additionally, although I know it's not a sufficient evidence, I compiled Nekopanda version using the not updated code by CUDA 8 compiler and compare the efficiency with Nekopanda's compiled files running same program but just replacing the dll file in the Amtsukaze folder, then I founed mine was not efficient as Nekopanda's . Since Nekopanda's AvsCUDA.dll and KFM.dll was both comiled half year later than the last updated code, I'm pretty sure there are code changes in the soure file. After all, in my guess Nekopanda's main and final purpose was not to create these filters, which were just tools he created to build his final program Amtsukaze of which I just find out the maintainance has been taken over by rigaya([url]https://github.com/rigaya/Amatsukaze/[/url]) though he/she didn't touch the AviSynthCUDAFilters codes.

If someone would compile the source code on Visualstudio2019, here is something might be of help to make older version CUDA compiler like CUDA8 work with Visualstudio2019/2017: https://www.olegtarasov.me/how-to-build-cuda-toolkit-projects-in-visual-studio-2017/ and https://www.pianshen.com/article/73121446097/.

**Comment by Feng-Cheng** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
